tool/gravity/cli/clusterupdate.go

```diff
@@ -96,6 +107,65 @@ func newClusterUpdater(
 	return updater, nil
 }
 
+// checkStatus returns an error if the cluster is degraded.
+// If force is true, warnings will be ignored.
+func checkStatus(ctx context.Context, force bool) error {
+	nodes, err := statusapi.NodeStatuses(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	var failedProbes []string
+	var warningProbes []string
+	for _, node := range nodes {
+		failedProbes = append(failedProbes, node.FailedProbes...)
+		warningProbes = append(warningProbes, node.WarnProbes...)
+	}
+
+	if len(failedProbes) > 0 {
+		return trace.BadParameter("unable to upgrade degraded cluster")
```

Comment:
I think we need to do a bit of wordsmithing to make these messages more friendly since they're gonna be shown to a user. For example, in case of critical probes:

```
Some cluster nodes are currently degraded and the upgrade is prohibited.
    // print degraded nodes
Please make sure the cluster is healthy before re-attempting the upgrade.
```